dist/formatters/utils/no-internal-summary-table-creator.js

```diff
@@ -7,6 +7,21 @@ const fs = require('fs');
 const process = require('process');
 const createAsciiTable = require('./create-ascii-table');
 
+/**
+ * Generates a summary table and optionally a CSV from linting messages for a specific rule.
+ * 
+ * This function processes an array of linting messages, filters them by a specified rule ID,
+ * and then generates a summary table and optionally a CSV file based on these messages. 
+ * 
+ * The summary table is always created and returned while the CSV file is created only if the 
+ * `createCSV` flag is set to `true`.
+ * 
+ * @param {Object[]} messages - An array of linting message objects to be processed.
+ * @param {string} ruleId - The ID of the linting rule to filter messages by.
+ * @param {boolean} [createCSV=true] - A boolean flag indicating whether to create a CSV file (`true`) 
+ *                                     or not. Defaults to `true`.
```

Comment:
should the default be `false`? 
why would everyone want a csv file always auto generated when they run lint?

Reply:
As I explained in my original PR here: https://github.com/iTwin/eslint-plugin/pull/76#issue-2411791921,

> The default behavior is still that the csv file is generated since that behavior currently exists in the released version of plugin. If any downstream apps rely on the existence of the csv file than that would make it a breaking change if we disabled the csv creation by default.

I do agree that the default behavior should be no csv but that version has already out there and I don't want to break people's expectations if that is something they are already to begin relying on. Lmk if I am overthinking this, I can make no csv default, I have no issues with that

Reply:
Sorry thought about this more after using it today. While breaking existing behavior, I think we are correcting it with what is the right way now, and they can opt back into this existing behavior themselves easily.

Reply:
ok that sounds good to me, it has been changed so that default behavior is no csv